# HG changeset patch
# User Rik <rik@octave.org>
# Date 1465228362 25200
#      Mon Jun 06 08:52:42 2016 -0700
# Node ID 590c39b94618b8f7b12e96938a8c724856e52d09
# Parent  2781d6efa88f381e5d3d1dffef0390345eacf638
Avoid duplicate naming of popen2 in octave::sys namespace (bug #47968).

* lo-sysdep.cc (win_popen2): Rename function from popen2.

* lo-sysdep.h (win_popen2): Rename function from popen2.

* syscalls.cc (popen2): Call win_popen2 if __WIN32__ and ! __CYGWIN__.

diff --git a/liboctave/system/lo-sysdep.cc b/liboctave/system/lo-sysdep.cc
--- a/liboctave/system/lo-sysdep.cc
+++ b/liboctave/system/lo-sysdep.cc
@@ -80,18 +80,18 @@ namespace octave
 #endif
 
       return gnulib::chdir (path.c_str ());
     }
 
 #if defined (__WIN32__) && ! defined (__CYGWIN__)
 
     pid_t
-    popen2 (const std::string& cmd, const string_vector& args,
-            bool sync_mode, int *fildes, std::string& msg)
+    win_popen2 (const std::string& cmd, const string_vector& args,
+                bool sync_mode, int *fildes, std::string& msg)
     {
       pid_t pid;
       PROCESS_INFORMATION pi;
       STARTUPINFO si;
       std::string command = "\"" + cmd + "\"";
       HANDLE hProcess = GetCurrentProcess ();
       HANDLE childRead, childWrite, parentRead, parentWrite;
       DWORD pipeMode;
diff --git a/liboctave/system/lo-sysdep.h b/liboctave/system/lo-sysdep.h
--- a/liboctave/system/lo-sysdep.h
+++ b/liboctave/system/lo-sysdep.h
@@ -35,30 +35,30 @@ namespace octave
 {
   namespace sys
   {
     extern std::string getcwd (void);
 
     extern int chdir (const std::string&);
 
 #if defined (__WIN32__) && ! defined (__CYGWIN__)
-    extern pid_t popen2 (const std::string&, const string_vector&,
-                         bool, int *, std::string&);
+    extern pid_t win_popen2 (const std::string&, const string_vector&,
+                             bool, int *, std::string&);
 #endif
   }
 }
 
 #if defined (OCTAVE_USE_DEPRECATED_FUNCTIONS)
 
 OCTAVE_DEPRECATED ("use 'octave::sys::getcwd' instead")
 const auto octave_getcwd = octave::sys::getcwd;
 
 OCTAVE_DEPRECATED ("use 'octave::sys::chdir' instead")
 const auto octave_chdir = octave::sys::chdir;
 
 #if defined (__WIN32__) && ! defined (__CYGWIN__)
 OCTAVE_DEPRECATED ("use 'octave::sys::popen2' instead")
-const auto octave_popen2 = octave::sys::popen2;
+const auto octave_popen2 = octave::sys::win_popen2;
 #endif
 
 #endif
 
 #endif
diff --git a/liboctave/system/oct-syscalls.cc b/liboctave/system/oct-syscalls.cc
--- a/liboctave/system/oct-syscalls.cc
+++ b/liboctave/system/oct-syscalls.cc
@@ -317,17 +317,17 @@ namespace octave
     }
 
     pid_t
     popen2 (const std::string& cmd, const string_vector& args,
             bool sync_mode, int *fildes, std::string& msg,
             bool &interactive)
     {
 #if defined (__WIN32__) && ! defined (__CYGWIN__)
-      return octave::sys::popen2 (cmd, args, sync_mode, fildes, msg);
+      return octave::sys::win_popen2 (cmd, args, sync_mode, fildes, msg);
 #else
       pid_t pid;
       int child_stdin[2], child_stdout[2];
 
       if (pipe (child_stdin, msg) == 0)
         {
           if (pipe (child_stdout, msg) == 0)
             {
